File: 03-strings/py/utf8.py
import sys

s = "Simpson"
mem_size = sys.getsizeof(s)
print(f"'{s}': length: {len(s)} chars: {len(s)} mem_size: {mem_size}")

# make bytes
print(s.encode('utf-8'))
print("after i:", s[2:])

#
print("===")

unicode = "Thanks 😊"
mem_size = sys.getsizeof(unicode)
print(f"'{unicode}': length: {len(unicode)} chars: {len(unicode)} mem_size: {mem_size}")

# make bytes
#
#
print(unicode.encode('utf-8'))
print("after h:", unicode[2:])

File: 03-strings/rs/src/utf8.rs
pub fn demo() {

    println!("===");
    println!("Demo UTF8");
    println!("===");

    let s = String::from("Simpson");
    println!(
        "'{}': length: {} chars: {} memsize: {}",
        s,
        s.len(),
        s.chars().count(),
        std::mem::size_of::<String>() + s.len()
    );
    println!("{:?}", s.as_bytes());
    println!("after Si: {:?}", &s[2..]);

    println!("===");

    let unicode = String::from("Thanks 😊");
    println!(
        "'{}': length: {} chars: {} memsize: {}",
        unicode,
        unicode.len(),
        unicode.chars().count(),
        std::mem::size_of::<String>() + unicode.len()
    );
    println!("{:?}", unicode.as_bytes());
    println!("after Th: {:?}", &unicode[2..]);
}